solution/backend/regulations/admin.py

```diff
@@ -1,10 +1,99 @@
-from django.contrib import admin
+import re
 
+from django.contrib import admin, messages
+from django.shortcuts import render
+from django.urls import path, reverse
+from django.http import HttpResponseRedirect
+from django.core.validators import URLValidator
+from django.core.exceptions import ValidationError
+
+import requests
 from solo.admin import SingletonModelAdmin
 
-from .models import SiteConfiguration
+from .models import SiteConfiguration, StatuteLinkConverter
 
 
 @admin.register(SiteConfiguration)
 class SiteConfigurationAdmin(SingletonModelAdmin):
     pass
+
+
+@admin.register(StatuteLinkConverter)
+class StatuteLinkConverterAdmin(admin.ModelAdmin):
+    change_list_template = "admin/import_conversions_button.html"
+    list_per_page = 200
+    search_fields = ["title", "section", "usc", "act", "source_url"]
+    ordering = ("title", "section", "usc", "act", "source_url")
+    fields = ("title", "section", "usc", "act", "source_url")
+    readonly_fields = ("source_url",)
+
+    def get_urls(self):
+        urls = super().get_urls()
+        my_urls = [
+            path('import_conversions/', self.show_import_conversions_page),
+        ]
+        return my_urls + urls
+
+    def import_conversions(self, text, url, act):
+        conversions = []
+        text = re.sub(r"</?[^>]+>", "", text)  # Strips HTML/XML tags from the response text
+        matches = re.findall(r"[Ss][Ee][Cc].?\s*(\d+).?\s*\[(\d+)\s*[Uu].?[Ss].?[Cc].?\s*(\w+)\]", text)  # Find all conversions
+        for section, title, usc in matches:
+            instance, created = self.model.objects.get_or_create(section=section, title=title, usc=usc, act=act, source_url=url)
+            if created:
+                conversions.append({
+                    "act": instance.act,
+                    "section": instance.section,
+                    "title": instance.title,
+                    "usc": instance.usc,
+                    "source_url": instance.source_url,
+                })
+        return conversions, len(matches)
+
+    def try_import(self, url, act):
+        if not url:
+            raise ValidationError("you must enter a URL!")
+        if not act:
+            raise ValidationError("you must enter an Act, for example \"SSA\"!")
+
+        try:
+            URLValidator(schemes=["http", "https"])(url)
+        except ValidationError:
+            raise ValidationError(f"{url} is not a valid URL!" if url else "you must enter a URL!")
+
+        response = requests.get(url)
+        response.raise_for_status()
+        conversions, matches = self.import_conversions(response.text, url, act)
+        if conversions:
+            return conversions
+        if matches and not conversions:
```

Comment:
and not conversions is not necessary.  If there was a conversion it would have already returned.

Reply:
Resolved in 2ea4ae6